packages/grafana-ui/src/components/ClickOutsideWrapper/ClickOutsideWrapper.tsx

```diff
@@ -26,21 +28,43 @@ export class ClickOutsideWrapper extends PureComponent<Props, State> {
     includeButtonPress: true,
     parent: typeof window !== 'undefined' ? window : null,
     useCapture: false,
+    setHandlers: true,
+    className: null,
   };
   myRef = createRef<HTMLDivElement>();
   state = {
     hasEventListener: false,
   };
 
   componentDidMount() {
+    if (this.props.setHandlers) {
+      this.setHandlers();
+    }
+  }
+
+  componentWillUnmount() {
+    this.removeHandlers();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.setHandlers) {
```

Comment:
If we were to move forward with this `setHandlers` prop, I think we would need  a check here that it doesn't already have listeners? 

It should compare prevProps to make sure it's false I think.

Reply:
Good catch! Added commit to properly handle props update.